src/server/journal/journal.cc

```diff
@@ -85,6 +85,7 @@ LSN Journal::GetLsn() const {
 
 void Journal::RecordEntry(TxId txid, Op opcode, DbIndex dbid, unsigned shard_cnt,
                           std::optional<cluster::SlotId> slot, Entry::Payload payload, bool await) {
+  lock_guard lk(state_mu_);
```

Comment:
We need this one and I think it's fine since it synchronizes with `Stop()`. Let me know @dranikpg if I am missing something here

Reply:
Yes, we should be synchronizing with .UnregisterOnChange. But I forgot about Close(), looks like that's also a potential trouble point 🙄

Reply:
I meant `Close()` above not `Stop`. Are you saying the lock there is problematic as well?

Reply:
it will be slow with this lock. we will have lock for every command

Reply:
It won' slow down anything. 

1. Without big value serialization it would be the whole shard unresponsive. So we are already `SLOW` on this path.
2. This `lock` will not be `held` unless we `preempt` when we write to the `journal`.
3. In fact this lock is not needed -- I was out of my head because I was running something locally incorrectly. We *might* preempt when we write to the log but this is fine (in fact I think I also caught a small corner case but I won't say much for now as I will need to ivnestigate)